### ISSUE TITLE
When pulling out layers from the model for extracting intermediate computation data, weights and activation output etc. the weights and biases are changing in different cases.

### ISSUE THREAD

**Issue by manp-git** [tensorflow/tensorflow#48824]
I realised is the weights and biases are getting modified in this procedure and graph structure is also changing when visualised through netron and the same changed structure is reflecting when checked via scripting to cross validate the netron output. Conv2_block1_2 for example had no BN layer before Relu. But If want to pull out the Relu output then it adds BN layers at every node which is being pulled out

outputs= [my_model.get_layer("conv2_block1_1_relu").output, my_model.get_layer("conv2_block1_2_conv").output, my_model.get_layer("conv2_block1_2_relu").output]

model_debug = tensorflow.keras.Model(inputs=my_model.inputs, outputs=outputs)

![orginal resnet tflite converted](https://user-images.githubusercontent.com/82860513/116591121-1deaef80-a93c-11eb-9473-e90b6a32abd4.png)
![with few pulled to outputs weights changed ](https://user-images.githubusercontent.com/82860513/116590970-f7c54f80-a93b-11eb-9bf6-0c0b8b270e19.png)
![with all initial layers pulled to outputs (2)](https://user-images.githubusercontent.com/82860513/116590996-feec5d80-a93b-11eb-81a3-71dec8623c28.png)

**Comment by UsharaniPagadala** [comment]
@manp-git

 Could you please fill the template
In order to expedite the trouble-shooting process, please provide a code snippet/colab gist to reproduce the issue reported here. Thanks!

**Comment by manp-git** [comment]
> @manp-git
> 
> Could you please fill the template
> In order to expedite the trouble-shooting process, please provide a code snippet/colab gist to reproduce the issue reported here. Thanks!


Hi @UsharaniPagadala , I have uploaded the snippets in the question. 1) The first one is default TF-Resnet-50 - quantisation to TFLite
2) Change in weights and lower graph when I pulled out : outputs= [my_model.get_layer("conv2_block1_1_relu").output, my_model.get_layer("conv2_block1_2_conv").output, my_model.get_layer("conv2_block1_2_relu").output]
3) change in entire graph when I pulled out all inital layers to output

**Comment by manp-git** [comment]
Update:- Solution I have figured out to weight values getting changed is that - not to output the block but only the relu layer. Why is that so, I dont know.
Secondly Why is the location of batch normalisation different in default resnet50 model as compared to tflite quantised resnet50 model, I am curious to know, as I want to change the weights to produce the relu output, which I am not able to do by set tensor, hence want to do using python convolution operations. What all keras functions would be required to reproduce the relu output from weights and output of previous layer is what i want to know.

**Comment by abattery** [comment]
Sorry, it is little hard to understand your questions since there are more than one questions and they are little broad. What is the major goal here?

**Comment by manp-git** [comment]
Hi, I have created a new issue for one of the things here: https://github.com/tensorflow/tensorflow/issues/48879

**Comment by tilakrayal** [comment]
@manp-git ,
Related issue #48879 has been closed.
Could you please update TensorFlow to the latest stable version v.2.7 and let us know if you are facing the same error. Thanks!

**Comment by manp-git** [comment]
Thanks for acting on this issue. I don't have access to that project anymore so won't be able to cross check.
Will work with the suggested version next time when required.
You may close the issue for now.

**Comment by tilakrayal** [comment]
@manp-git ,
Please feel free to move this issue to closed status.Thanks
